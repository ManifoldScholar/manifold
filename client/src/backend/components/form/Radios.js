import React, { Component } from "react";
import PropTypes from "prop-types";
import { UID } from "react-uid";
import GlobalForm from "global/components/form";
import Option from "./Radio/Option";
import RadioLabel from "./Radio/Label";
import classnames from "classnames";
import isString from "lodash/isString";
import Instructions from "./Instructions";
import withFormOptions from "hoc/with-form-options";

class FormRadios extends Component {
  static displayName = "Form.Radios";

  static propTypes = {
    options: PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.any.isRequired,
        label: PropTypes.string.isRequired,
        instructions: PropTypes.string,
        internalValue: PropTypes.any.isRequired
      })
    ).isRequired,
    label: PropTypes.string,
    prompt: PropTypes.string,
    inline: PropTypes.bool,
    name: PropTypes.string,
    value: PropTypes.any,
    set: PropTypes.func,
    inputClasses: PropTypes.string,
    instructions: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    focusOnMount: PropTypes.bool
  };

  static defaultProps = {
    focusOnMount: false
  };

  get focusOnMount() {
    return this.props.focusOnMount;
  }

  get options() {
    return this.props.options;
  }

  get optionProps() {
    return {
      inline: this.props.inline,
      onChange: this.props.onChange,
      value: this.props.value
    };
  }

  get inputClasses() {
    return classnames(this.props.inputClasses, {
      "form-input": true,
      "form-input-radios": true,
      "extra-space-bottom": true,
      wide: this.props.wide
    });
  }

  get idPrefix() {
    return "radios";
  }

  get idForErrorPrefix() {
    return "radios-error";
  }

  get idForInstructionsPrefix() {
    return "radios-instructions";
  }

  render() {
    return (
      <UID>
        {id => (
          <GlobalForm.Errorable
            className={this.inputClasses}
            name={this.props.name}
            errors={this.props.errors}
            label={this.props.label}
            idForError={`${this.idForErrorPrefix}-${id}`}
          >
            <fieldset
              className="form-input-radios__wrapper"
              aria-describedby={`${this.idForErrorPrefix}-${id} ${
                this.idForInstructionsPrefix
              }-${id}`}
            >
              <RadioLabel
                label={this.props.label}
                prompt={this.props.prompt}
                hasInstructions={isString(this.props.instructions)}
              />
              <Instructions
                instructions={this.props.instructions}
                id={`${this.idForInstructionsPrefix}-${id}`}
              />
              {this.options.map((option, index) => (
                <Option
                  key={`${this.idPrefix}-${id}-${option.internalValue}`}
                  option={option}
                  focusOnMount={this.focusOnMount && index === 0}
                  tabIndex={index === 0 ? 0 : -1}
                  groupName={`${this.idPrefix}-${id}`}
                  {...this.optionProps}
                />
              ))}
            </fieldset>
          </GlobalForm.Errorable>
        )}
      </UID>
    );
  }
}

export default withFormOptions(FormRadios);
