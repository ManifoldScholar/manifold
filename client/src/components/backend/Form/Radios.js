import React, { Component, PropTypes } from 'react';
import sharedPropsValidation from './propTypes';
import { Form } from 'components/backend';
import classNames from 'classnames';

export default class FormRadios extends Component {

  static displayName = "Form.Radios";

  static propTypes = {
    ...sharedPropsValidation,
    options: PropTypes.arrayOf(PropTypes.shape({
      value: PropTypes.any.isRequired,
      label: PropTypes.string.isRequired
    })).isRequired
  };

  static defaultProps = {
    layout: "horizontal"
  };

  render() {

    const inputClassNames = classNames('form-toggle', 'radio', this.props.layout);

    return (
      <div className="form-input">
        <label>{this.props.label}</label>
        {this.props.options.map((option) => {
          return (
            <label htmlFor={option.value} className={inputClassNames} key={option.value} >
              <Form.Connect.Set {...this.props} value={option.value} >
                <input id={option.value} checked={this.props.value === option.value} type="radio" />
              </Form.Connect.Set>
              <span className="toggle-indicator"></span>
              <span className="toggle-label">{option.label}</span>
            </label>
          );
        })}

      </div>
    );
  }

}
