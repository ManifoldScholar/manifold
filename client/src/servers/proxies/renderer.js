import config from "config";
import proxy from "http-proxy-middleware";
import ch from "../../helpers/consoleHelpers";
import createStore from "../../store/createStore";
import exceptionRenderer from "../../helpers/exceptionRenderer";
import manifoldBootstrap from "bootstrap";
import CookieHelper from "helpers/cookie/Server";

const ssrRenderUrl = `http://${config.services.client.domain}:${config.services.client.sparePort}`;

export default function makeRendererProxy(requestHandler) {
  const reactServerProxy = proxy({
    target: ssrRenderUrl,
    changeOrigin: true,
    logLevel: "silent",

    onError: (err, req, res) => {
      ch.error(`Error proxying to port ${ssrRenderUrl}`);
      ch.error(`Perhaps the SSR service is in the process of reloading?`);
      ch.error(`Falling back to SSR rescue service only`);

      const store = createStore();

      const render = () => {
        try {
          requestHandler(req, res);
        } catch (error) {
          if (error.code === "MODULE_NOT_FOUND") {
            const msg =
              "Waiting for initial Webpack build to complete. Wait a few seconds " +
              "and reload.";
            ch.error(msg);
            res.setHeader("Content-Type", "text/html");
            res.end(msg);
          } else {
            ch.error(
              `Rendering fallback failed to render inn server-developmet.js`
            );
            res.setHeader("Content-Type", "text/html");
            res.end(exceptionRenderer(error));
          }
        }
      };

      const promises = [];
      const cookie = new CookieHelper(req, res);
      promises.push(manifoldBootstrap(store.getState, store.dispatch, cookie));
      Promise.all(promises).then(render, render);
    }
  });
  return reactServerProxy;
}
