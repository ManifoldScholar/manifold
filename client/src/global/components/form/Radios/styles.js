import styled from "@emotion/styled";

export const RadiosWrapper = styled.fieldset`
  padding: 0;
  margin: 0;
  border: none;
`;
