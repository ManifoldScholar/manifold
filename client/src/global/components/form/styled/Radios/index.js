import React, { Component } from "react";
import PropTypes from "prop-types";
import { UIDConsumer } from "react-uid";
import Errorable from "global/components/form/Errorable";
import Option from "../Radio/Option";
import RadioLabel from "../Radio/Label";
import classnames from "classnames";
import isString from "lodash/isString";
import Instructions from "../Instructions";
import withFormOptions from "hoc/withFormOptions";
import * as Styled from "./styles";

class FormRadios extends Component {
  static displayName = "Form.Radios";

  static propTypes = {
    options: PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.any.isRequired,
        label: PropTypes.string.isRequired,
        instructions: PropTypes.string
      })
    ).isRequired,
    label: PropTypes.string,
    prompt: PropTypes.string,
    inline: PropTypes.bool,
    name: PropTypes.string,
    value: PropTypes.any,
    set: PropTypes.func,
    beforeOnChange: PropTypes.func,
    inputClasses: PropTypes.string,
    instructions: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    focusOnMount: PropTypes.bool
  };

  static defaultProps = {
    focusOnMount: false
  };

  get focusOnMount() {
    return this.props.focusOnMount;
  }

  get options() {
    return this.props.options;
  }

  get optionProps() {
    return {
      inline: this.props.inline,
      onChange: this.props.onChange,
      value: this.props.value
    };
  }

  get inputClasses() {
    return classnames(this.props.inputClasses, {
      "extra-space-bottom": true,
      wide: this.props.wide
    });
  }

  get idPrefix() {
    return "radios";
  }

  get idForErrorPrefix() {
    return "radios-error";
  }

  get idForInstructionsPrefix() {
    return "radios-instructions";
  }

  render() {
    return (
      <UIDConsumer>
        {id => (
          <Errorable
            className={this.inputClasses}
            name={this.props.name}
            errors={this.props.errors}
            label={this.props.label}
            idForError={`${this.idForErrorPrefix}-${id}`}
          >
            <Styled.RadiosWrapper
              aria-describedby={`${this.idForErrorPrefix}-${id} ${this.idForInstructionsPrefix}-${id}`}
            >
              <RadioLabel
                label={this.props.label}
                prompt={this.props.prompt}
                hasInstructions={isString(this.props.instructions)}
              />
              <Instructions
                instructions={this.props.instructions}
                id={`${this.idForInstructionsPrefix}-${id}`}
              />
              {this.options.map((option, index) => (
                <Option
                  key={option.key}
                  option={option}
                  focusOnMount={this.focusOnMount && index === 0}
                  groupName={`${this.idPrefix}-${id}`}
                  {...this.optionProps}
                />
              ))}
            </Styled.RadiosWrapper>
          </Errorable>
        )}
      </UIDConsumer>
    );
  }
}

export default withFormOptions(FormRadios);
