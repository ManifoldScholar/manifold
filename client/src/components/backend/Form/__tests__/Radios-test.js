import React from 'react';
import renderer from 'react-test-renderer';
import Radios from '../Radios';

describe("Backend.Form.Radios component", () => {

  const changeMock = jest.fn();
  const options = [
    {
      label: "option-1",
      value: "1"
    },
    {
      label: "option-2",
      value: "2"
    }
  ];

  it('renders correctly', () => {
    const component = renderer.create(
      <Radios
        options={options}
        label="Label this"
        name="attributes[fake]"
        onChange={changeMock}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
