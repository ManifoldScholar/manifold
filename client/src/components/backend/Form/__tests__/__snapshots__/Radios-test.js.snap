// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backend.Form.Radios component renders correctly 1`] = `
<div
  className="form-input"
>
  <label
    style={
      Object {
        "marginBottom": "1.5em",
      }
    }
  >
    Label this
  </label>
  <label
    className="form-toggle radio"
    htmlFor="1"
  >
    <input
      checked={false}
      id="1"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <span
      aria-hidden="true"
      className="toggle-indicator"
    />
    <span
      className="toggle-label"
    >
      option-1
    </span>
  </label>
  <label
    className="form-toggle radio"
    htmlFor="2"
  >
    <input
      checked={false}
      id="2"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <span
      aria-hidden="true"
      className="toggle-indicator"
    />
    <span
      className="toggle-label"
    >
      option-2
    </span>
  </label>
</div>
`;
